File: php_current/library/Adsolut/Http/Response.php
<?php

/**
 * Description of Response
 *
 * @author marcello
 */
class Adsolut_Http_Response {

    private $errors = array();
    private $data;

    function __construct($raw_response) {
        $this->setErrors($raw_response->errors);
        $this->setData($raw_response->data);
    }

    public function hasErrors() {
        return !empty($this->errors);
    }

    private function setErrors($errors) {
        $this->errors = $errors;
    }

    private function setData($data) {
        $this->data = $data;
    }

    public function getErrors() {
        return $this->errors;
    }

    public function getData() {
        return $this->data;
    }

}

File: java/src/main/java/it/adsolut/bbitaly/response/Response.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package it.adsolut.bbitaly.response;

import java.util.ArrayList;
import java.util.List;
import javax.annotation.Resource;
import org.dozer.DozerBeanMapper;
import org.springframework.validation.ObjectError;

/**
 *
 * @author marcello
 */
public abstract class Response<T,K> {

    @Resource(name = "dozermapper")
    protected DozerBeanMapper dozermapper;
    
    private List<String> errors = new ArrayList<String>();
    
    protected K data;

    public Response(DozerBeanMapper dozermapper) {
        this.dozermapper = dozermapper;
    }

    public List<String> getErrors() {
        return errors;
    }

    public void setErrors(List<String> errors) {
        this.errors = errors;
    }

    public void addError(String code) {
        errors.add(code);
    }

    public void addError(List<ObjectError> errors) {
        for (ObjectError oe : errors) {
            this.addError(oe.getDefaultMessage());
        }
    }

    public K getData(){
        return this.data;
    }
    
    public void setData(T data){
        this.data = map(data);
    }
    
    protected abstract K map(T data);
}
